editors/vscode/client/extension.ts

```diff
@@ -90,10 +92,45 @@ export async function activate(context: ExtensionContext) {
   const outputChannel = window.createOutputChannel(outputChannelName);
   const traceOutputChannel = window.createOutputChannel(traceOutputChannelName);
 
-  const ext = process.platform === 'win32' ? '.exe' : '';
-  // NOTE: The `./target/release` path is aligned with the path defined in .github/workflows/release_vscode.yml
-  const command = process.env.SERVER_PATH_DEV ??
-    join(context.extensionPath, `./target/release/oxc_language_server${ext}`);
+  async function findBinary(): Promise<string> {
+    const cfg = workspace.getConfiguration('oxc');
+
+    let bin = cfg.get<string>('binPath', '');
+    if (bin) {
+      try {
+        await fsPromises.access(bin);
+        return bin;
+      } catch {}
+    }
+
+    const workspaceFolders = workspace.workspaceFolders;
+    if (workspaceFolders) {
+      try {
+        return await Promise.any(
+          workspaceFolders.map(async (folder) => {
+            const binPath = join(
+              folder.uri.fsPath,
+              'node_modules',
+              '.bin',
+              'oxc_language_server',
+            );
+
+            await fsPromises.access(binPath);
```

Comment:
If any of these throw, the whole Promise.all will reject. You will need to handle errors for each individual promise, or use allSettled

Reply:
No, I'm using `Promise.any` instead of `Promise.all`.